🐛 Bug Report: Cannot find module '@backstage/plugin-scaffolder-backend-module-gitlab'

### 📜 Description

I installed the plugin-scaffolder-backend-module-gitlab as suggested in https://github.com/backstage/backstage/tree/master/plugins/scaffolder-backend-module-gitlab. 

When I run `yarn dev`, it gives an error "Cannot find module '@backstage/plugin-scaffolder-backend-module-gitlab'"

### 👍 Expected behavior

Application should work.

### 👎 Actual Behavior with Screenshots

```
C:\DEV\Playground\backstage\src>yarn dev
yarn run v1.22.19
$ concurrently "yarn start" "yarn start-backend"
$ yarn workspace app start
$ yarn workspace backend start
$ backstage-cli package start
$ backstage-cli package start
[1] Build succeeded
[0] Loaded config from app-config.yaml
[0] [webpack-dev-server] Project is running at:
[0] [webpack-dev-server] Loopback: http://localhost:3000/, http://127.0.0.1:3000/
[0]  [webpack-dev-server] Content not from webpack is served from 'C:\DEV\Playground\backstage\src\packages\app\public' directory
[0] [webpack-dev-server] 404s will fallback to '/index.html'
[0]  [webpack-dev-middleware] wait until bundle finished: /
[1] C:\DEV\Playground\backstage\src\packages\backend\dist\main.js:499
[1] /\*\*\*\*\*\*/                    throw e;
[1]                             ^
[1]
[1] Error: Cannot find module '@backstage/plugin-scaffolder-backend-module-gitlab'
[1]     at webpackMissingModule (webpack-internal:///./src/plugins/scaffolder.ts:11:50)
[1]     at eval (webpack-internal:///./src/plugins/scaffolder.ts:11:175)
[1]     at Module../src/plugins/scaffolder.ts (C:\DEV\Playground\backstage\src\packages\backend\dist\main.js:118:1)
[1]     at __webpack_require__ (C:\DEV\Playground\backstage\src\packages\backend\dist\main.js:496:33)
[1]     at fn (C:\DEV\Playground\backstage\src\packages\backend\dist\main.js:656:21)
[1]     at eval (webpack-internal:///./src/index.ts:11:77)
[1]     at Module../src/index.ts (C:\DEV\Playground\backstage\src\packages\backend\dist\main.js:30:1)
[1]     at __webpack_require__ (C:\DEV\Playground\backstage\src\packages\backend\dist\main.js:496:33)
[1]     at C:\DEV\Playground\backstage\src\packages\backend\dist\main.js:1475:37
[1]     at Object.<anonymous> (C:\DEV\Playground\backstage\src\packages\backend\dist\main.js:1477:12)
[1]     at Module._compile (node:internal/modules/cjs/loader:1126:14)
[1]     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
[1]     at Module.load (node:internal/modules/cjs/loader:1004:32)
[1]     at Function.Module._load (node:internal/modules/cjs/loader:839:12)
[1]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
[1]     at node:internal/main/run_main_module:17:47 {
[1]   code: 'MODULE_NOT_FOUND'
[1] }
[0] webpack compiled successfully
```

### 👟 Reproduction steps

Follow installation instructions at https://github.com/backstage/backstage/tree/master/plugins/scaffolder-backend-module-gitlab

Add plugin with following command:

yarn add --cwd packages/backend @backstage/plugin-scaffolder-backend-module-gitlab

Then in packages/backend/src/plugins/scaffolder.ts

```
import {CatalogClient} from '@backstage/catalog-client';
import {createBuiltinActions, createRouter} from '@backstage/plugin-scaffolder-backend';
import {Router} from 'express';
import type {PluginEnvironment} from '../types';
import {ScmIntegrations} from "@backstage/integration";
import {
  createGitlabGroupEnsureExistsAction,
  createGitlabProjectAccessTokenAction,
  createGitlabProjectDeployTokenAction,
  createGitlabProjectVariableAction
} from "@backstage/plugin-scaffolder-backend-module-gitlab";

export default async function createPlugin(
    env: PluginEnvironment,
): Promise<Router> {
  const catalogClient = new CatalogClient({
    discoveryApi: env.discovery,
  });

  const integrations = ScmIntegrations.fromConfig(env.config)

  // Create BuiltIn Actions
  const builtInActions = createBuiltinActions({
    integrations,
    catalogClient,
    config: env.config,
    reader: env.reader,
  });

// Add Gitlab Actions
  const actions = [
    ...builtInActions,
    createGitlabProjectAccessTokenAction({integrations: integrations}),
    createGitlabProjectDeployTokenAction({integrations: integrations}),
    createGitlabProjectVariableAction({integrations: integrations}),
    createGitlabGroupEnsureExistsAction({integrations: integrations}),
  ];

  return await createRouter({
    logger: env.logger,
    config: env.config,
    database: env.database,
    reader: env.reader,
    catalogClient,
    actions,
    identity: env.identity,
  });
}

```

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

$ C:\DEV\Playground\backstage\src\node_modules\.bin\backstage-cli info
OS:   Windows_NT 10.0.19044 - win32/x64
node: v16.17.1
yarn: 1.22.19
cli:  0.22.7 (installed)
backstage:  1.14.0

Dependencies:
  @backstage/app-defaults                            1.3.1
  @backstage/backend-app-api                         0.4.3
  @backstage/backend-common                          0.18.5
  @backstage/backend-dev-utils                       0.1.1
  @backstage/backend-plugin-api                      0.5.2
  @backstage/backend-tasks                           0.5.2
  @backstage/catalog-client                          1.4.1
  @backstage/catalog-model                           1.3.0
  @backstage/cli-common                              0.1.12
  @backstage/cli-node                                0.1.0
  @backstage/cli                                     0.22.7
  @backstage/config-loader                           1.3.0
  @backstage/config                                  1.0.7
  @backstage/core-app-api                            1.8.0
  @backstage/core-components                         0.12.5, 0.13.1
  @backstage/core-plugin-api                         1.5.1
  @backstage/errors                                  1.1.5
  @backstage/eslint-plugin                           0.1.3
  @backstage/integration-aws-node                    0.1.3
  @backstage/integration-react                       1.1.13
  @backstage/integration                             1.4.5
  @backstage/plugin-api-docs                         0.9.3
  @backstage/plugin-app-backend                      0.3.45
  @backstage/plugin-auth-backend                     0.18.3
  @backstage/plugin-auth-node                        0.2.14
  @backstage/plugin-catalog-backend-module-gitlab    0.2.1
  @backstage/plugin-catalog-backend                  1.9.1
  @backstage/plugin-catalog-common                   1.0.13
  @backstage/plugin-catalog-graph                    0.2.30
  @backstage/plugin-catalog-import                   0.9.8
  @backstage/plugin-catalog-node                     1.3.6
  @backstage/plugin-catalog-react                    1.6.0
  @backstage/plugin-catalog                          1.11.0
  @backstage/plugin-github-actions                   0.5.18
  @backstage/plugin-org                              0.6.8
  @backstage/plugin-permission-backend               0.5.20
  @backstage/plugin-permission-common                0.7.5
  @backstage/plugin-permission-node                  0.7.8
  @backstage/plugin-permission-react                 0.4.12
  @backstage/plugin-proxy-backend                    0.2.39
  @backstage/plugin-scaffolder-backend-module-gitlab 0.2.0
  @backstage/plugin-scaffolder-backend               1.14.0
  @backstage/plugin-scaffolder-common                1.3.0
  @backstage/plugin-scaffolder-node                  0.1.3
  @backstage/plugin-scaffolder-react                 1.4.0
  @backstage/plugin-scaffolder                       1.13.1
  @backstage/plugin-search-backend-module-catalog    0.1.1
  @backstage/plugin-search-backend-module-pg         0.5.6
  @backstage/plugin-search-backend-module-techdocs   0.1.1
  @backstage/plugin-search-backend-node              1.2.1
  @backstage/plugin-search-backend                   1.3.1
  @backstage/plugin-search-common                    1.2.3
  @backstage/plugin-search-react                     1.6.0
  @backstage/plugin-search                           1.3.0
  @backstage/plugin-sonarqube-backend                0.1.10
  @backstage/plugin-sonarqube-react                  0.1.5
  @backstage/plugin-sonarqube                        0.6.7
  @backstage/plugin-tech-radar                       0.6.4
  @backstage/plugin-techdocs-backend                 1.6.2
  @backstage/plugin-techdocs-module-addons-contrib   1.0.13
  @backstage/plugin-techdocs-node                    1.7.1
  @backstage/plugin-techdocs-react                   1.1.6
  @backstage/plugin-techdocs                         1.6.2
  @backstage/plugin-user-settings                    0.7.3
  @backstage/release-manifests                       0.0.9
  @backstage/test-utils                              1.3.1
  @backstage/theme                                   0.2.19, 0.3.0
  @backstage/types                                   1.0.2
  @backstage/version-bridge                          1.0.4
Done in 1.18s.


### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

--- a/.changeset/few-games-swim.md
+++ b/.changeset/few-games-swim.md
@@ -1,0 +1,6 @@
+---
+'@backstage/plugin-cicd-statistics-module-gitlab': patch
+'@backstage/plugin-scaffolder-backend-module-gitlab': patch
+---
+
+Fixed publish configuration.

--- a/plugins/cicd-statistics-module-gitlab/package.json
+++ b/plugins/cicd-statistics-module-gitlab/package.json
@@ -7,8 +7,7 @@
   "license": "Apache-2.0",
   "publishConfig": {
     "access": "public",
-    "main": "dist/index.cjs.js",
-    "module": "dist/index.esm.js",
+    "main": "dist/index.esm.js",
     "types": "dist/index.d.ts"
   },
   "backstage": {

--- a/plugins/scaffolder-backend-module-gitlab/package.json
+++ b/plugins/scaffolder-backend-module-gitlab/package.json
@@ -6,7 +6,7 @@
   "license": "Apache-2.0",
   "publishConfig": {
     "access": "public",
-    "main": "dist/index.esm.js",
+    "main": "dist/index.cjs.js",
     "types": "dist/index.d.ts"
   },
   "backstage": {

